src/test/java/com/cdancy/bitbucket/rest/features/TasksApiMockTest.java

```diff
@@ -97,11 +105,96 @@ public void testGetTask() throws Exception {
+            assertThat(instanceNowResolved.state()).isEqualTo(taskResolved);
+
+            server.takeRequest();
+            final RecordedRequest putRequest = server.takeRequest();
+            assertThat(putRequest.getRequestLine()).isEqualTo(String.format("PUT %s/%s HTTP/1.1", tasksEndpoint, taskId));
+            final String json = payloadFromResource("/task-anchor-resolved.json");
+            assertThat(putRequest.getBody().readUtf8()).isEqualTo(json);
+        } finally {
+            baseApi.close();
+            server.shutdown();
+        }
+    }
+
+
+    public void testOpenTask() throws Exception {
+        final MockWebServer server = mockWebServer();
+
+        server.enqueue(new MockResponse().setBody(payloadFromResource(taskJson)).setResponseCode(200));
+        server.enqueue(new MockResponse().setBody(payloadFromResource("/task-open.json")).setResponseCode(201));
+        final BitbucketApi baseApi = api(server.getUrl("/"));
+        final TasksApi api = baseApi.tasksApi();
+        try {
+            final TaskAnchor taskAnchor = api.get(taskId).anchor();
+            final UpdateTask updateTaskOpen = UpdateTask.update(taskAnchor, taskId, taskOpen, 1, 1);
+
+            final Task instanceNowOpen = api.update(updateTaskOpen.id(), updateTaskOpen);
+            assertThat(instanceNowOpen).isNotNull();
+            assertThat(instanceNowOpen.errors().isEmpty()).isTrue();
+            assertThat(instanceNowOpen.state()).isEqualTo(taskOpen);
+
+            server.takeRequest();
+            final RecordedRequest putRequest = server.takeRequest();
+            assertThat(putRequest.getRequestLine()).isEqualTo(String.format("PUT %s/%s HTTP/1.1", tasksEndpoint, taskId));
+            final String json = payloadFromResource("/task-anchor-open.json");
+            assertThat(putRequest.getBody().readUtf8()).isEqualTo(json);
+        } finally {
+            baseApi.close();
+            server.shutdown();
+        }
+    }
+
+    public void testOpenTaskOnErrors() throws Exception {
+        final MockWebServer server = mockWebServer();
+
+        server.enqueue(new MockResponse().setBody(payloadFromResource(taskJson)).setResponseCode(200));
+        server.enqueue(new MockResponse().setBody(payloadFromResource(errorsJson)).setResponseCode(404));
+        final BitbucketApi baseApi = api(server.getUrl("/"));
+        final TasksApi api = baseApi.tasksApi();
+        try {
+
+            final TaskAnchor taskAnchor = api.get(taskId).anchor();
+            final UpdateTask updateTaskOpen = UpdateTask.update(taskAnchor, taskId, taskOpen, 1, 1);
+
+            final Task instanceNowOpen = api.update(updateTaskOpen.id(), updateTaskOpen);
+            assertThat(instanceNowOpen).isNotNull();
+            assertThat(instanceNowOpen.errors().isEmpty()).isFalse();
+            assertThat(instanceNowOpen.anchor()).isNull();
+            assertThat(instanceNowOpen.state()).isNull();
+
+            server.takeRequest();
```

Comment:
I needed to access the second _RecordedRequest_ so just throwing away the first one above because it is already being tested elsewhere. There did not seem to be a way to access the second one directly and I did not want to make a test class wide _Task_ field because that breaks down test isolation. This seemed the lesser of two evils.

Reply:
Fair enough.